core/trino-main/src/main/java/io/trino/util/StatementUtils.java

```diff
@@ -327,4 +381,21 @@ public Class<? extends DataDefinitionTask<T>> getTaskType()
             return taskType.orElseThrow(() -> new IllegalStateException(queryType + " does not have a task type"));
         }
     }
+
+    private static HashSet<Table> getTables(io.trino.sql.tree.Node node, HashSet<String> withTablesToExclude)
+    {
+        HashSet<Table> queryTables = new HashSet<Table>();
```

Comment:
This is a HashSet<Table>, but below you call `add(String)` and `contains(string)`, so it should probably be a `HashSet<String>` or you should pass `Table`s

Reply:
There are 2 HashSets.  The one that is returned is a collection of `Table`s.  The second HashSet of strings is passed in the recursive calls to keep track of CTE aliases which are also `Table` nodes but not actual source tables.  These are removed from the final set of `Table`s returned.

Reply:
Yep sorry - read through too quickly